openedx/core/djangoapps/embargo/api.py

```diff
@@ -170,10 +170,11 @@ def _country_code_from_ip(ip_addr):
         str: A 2-letter country code.
 
     """
-    if ip_addr.find(':') >= 0:
-        return pygeoip.GeoIP(settings.GEOIPV6_PATH).country_code_by_addr(ip_addr)
-    else:
-        return pygeoip.GeoIP(settings.GEOIP_PATH).country_code_by_addr(ip_addr)
+    reader = geoip2.database.Reader(settings.GEOIP_PATH)
+
+    response = reader.country(ip_addr)
```

Comment:
In testing this out on existing IP addresses used by users, I found that some calls result in an exception being thrown.

def new_country_code_from_ip(ip_addr):
    reader = geoip2.database.Reader(NEW_GEOIP_PATH)
    try:
        response = reader.country(ip_addr)
        country_code = response.country.iso_code
    except geoip2.errors.AddressNotFoundError as exc:
        country_code = ""   # Or None or whatever is appropriate.
    return country_code